android/src/main/java/com/guichaguri/trackplayer/service/models/Track.java

```diff
@@ -88,6 +97,7 @@ public Track(Context context, Bundle bundle, int ratingType) {
 
         userAgent = bundle.getString("userAgent");
         artwork = Utils.getUri(context, bundle, "artwork");
+        icymetadata = bundle.getBoolean("icymetadata");
```

Comment:
Why not add one new track `type` value?

Reply:
In HLS it is possible to obtain a metadata. It is then not influenced by the `type` of transmission.

Reply:
I mean, a HLS stream can't be an icy stream and an icy stream can't be a HLS stream. Both protocols are not supported together, so it can only be one or another.